dotnet/src/Functions/Functions.Prompty/Extensions/PromptyKernelExtension.cs

```diff
@@ -0,0 +1,127 @@
+using Microsoft.SemanticKernel.Prompty.Core;
+using YamlDotNet.Serialization;
+
+namespace Microsoft.SemanticKernel.Prompty.Extension;
+
+/// <summary>
+/// extension methods for <see cref="Kernel"/> to create a <see cref="KernelFunction"/> from a prompty file.
+/// </summary>
+public static class PromptyKernelExtension
+{
+    /// <summary>
+    /// Create a <see cref="KernelFunction"/> from a prompty file.
+    /// </summary>
+    /// <param name="_">kernel</param>
+    /// <param name="promptyPath">path to prompty file.</param>
+    /// <param name="promptTemplateFactory">prompty template factory, if not provided, a <see cref="LiquidPromptTemplateFactory"/> will be used.</param>
+    /// <param name="loggerFactory">logger factory</param>
+    /// <returns><see cref="KernelFunction"/></returns>
+    /// <exception cref="ArgumentNullException"></exception>
+    /// <exception cref="NotSupportedException"></exception>
+    public static KernelFunction CreateFunctionFromPrompty(
+        this Kernel _,
+        string promptyPath,
+        IPromptTemplateFactory? promptTemplateFactory = null,
+        ILoggerFactory? loggerFactory = null)
+    {
+        var text = File.ReadAllText(promptyPath);
+
+        promptTemplateFactory ??= new LiquidPromptTemplateFactory(); // use liquid template factory by default
+
+        // create PromptTemplateConfig from text
+        // step 1
+        // retrieve the header, which is in yaml format and put between ---
+        //
+        // e.g
+        // file: chat.prompty
+        // ---
+        // name: Contoso Chat Prompt
+        // description: A retail assistant for Contoso Outdoors products retailer.
+        // authors:
+        //   - XXXX
+        // model:
+        //   api: chat
+        //   configuration:
+        //     type: azure_openai
+        //     azure_deployment: gpt - 35 - turbo
+        //     api_version: 2023 - 07 - 01 - preview
+        //   parameters:
+        //     tools_choice: auto
+        //     tools:
+        //     -type: function
+        //       function:
+        //         name: test
+        //         description: test function
+        //         parameters:
+        //           properties:
+        //             location:
+        //               description: The city and state or city and country, e.g.San Francisco, CA
+        //                 or Tokyo, Japan
+        // ---
+        // ... (rest of the prompty content)
+
+        var splits = text.Split(["---"], StringSplitOptions.RemoveEmptyEntries);
+        var yaml = splits[0];
+        var content = splits[1];
+
+        var deserializer = new DeserializerBuilder().Build();
+        var prompty = deserializer.Deserialize<PromptyYaml>(yaml);
+
+        // step 2
+        // create a prompt template config from the prompty object
+        var promptTemplateConfig = new PromptTemplateConfig
+        {
+            Name = prompty.Name, // TODO: sanitize name
```

Comment:
What sanitization is needed?

Reply:
The `prompty.Name` comes from prompty file and may contain space/special characters etc

Reply:
as far as the spec is concerned, the name is strictly considered metadata - if we need to extract a useful function name, my sense is that the filename (sans extension) would be a better candidate.